sdk/test-utils/recorder-new/MIGRATION.md

```diff
@@ -0,0 +1,264 @@
+# Migration Guide
+
+This document outlines key differences between the legacy recorder and the new Unified Recorder client. The Unified Recorder replaces the existing `nock/nise`-based recorder with a solution that uses the language-agnostic [test proxy server].
+
+## Prerequisites
+
+- [Docker] is required, as the [test proxy server] is run in a container during testing.
```

Comment:
I'm running within WSL and I had to do 2 additional steps to get the proxy-tool working.

1. Start the docker daemon after installing docker `sudo service docker start`
2. Add myself to the docker user group `sudo usermod -aG docker ${USER}` (then log out and in again)

Reply:
@timovv is it a good idea to refer this https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/test-utils/recorder-new/README.md#with-the-docker-run-command ?

Mention something like "For some reason the dev-tool command did not work for you or the internal docker run step is failing.. try running docker run command manually to see what's going wrong and reach out to us if you're blocked"

Reply:
> Mention something like "For some reason the dev-tool command did not work for you or the internal docker run step is failing.. try running docker run command manually to see what's going wrong and reach out to us if you're blocked"

Worth mentioning something like that, sure.

Regarding your issue, @joheredi, do you think it would suffice to just say 'ensure the Docker daemon is running and you have permission to use it', which covers the basics -- the actual steps you'd have to take would vary a fair bit depending on what environment you're on, so am hesitant to get too specific.

Reply:
@timovv I was thinking in similar lines, maybe mention Jose's steps in the troubleshooting guide? at the end.

Reply:
It would also be helpful if dev-tool gave a helpful error message when the test proxy fails to start due to Docker issues, since at the moment it just exits and doesn't print anything useful. I'll raise an issue.

Reply:
Added a quick note on this in the prerequisites section